recipes/libarchive/all/conanfile.py

```diff
@@ -116,110 +116,68 @@ def requirements(self):
-        self._cmake.definitions["ENABLE_LZO"] = self.options.with_lzo
-        self._cmake.definitions["ENABLE_LZMA"] = self.options.with_lzma
-        self._cmake.definitions["ENABLE_ZSTD"] = self.options.with_zstd
-        self._cmake.definitions["ENABLE_ZLIB"] = self.options.with_zlib
-        self._cmake.definitions["ENABLE_BZip2"] = self.options.with_bzip2
+        tc.variables["ENABLE_NETTLE"] = self.options.with_nettle
+        tc.variables["ENABLE_OPENSSL"] = self.options.with_openssl
+        tc.variables["ENABLE_LIBB2"] = self.options.with_libb2
+        tc.variables["ENABLE_LZ4"] = self.options.with_lz4
+        tc.variables["ENABLE_LZO"] = self.options.with_lzo
+        tc.variables["ENABLE_LZMA"] = self.options.with_lzma
+        tc.variables["ENABLE_ZSTD"] = self.options.with_zstd
+        tc.variables["ENABLE_ZLIB"] = self.options.with_zlib
+        tc.variables["ENABLE_BZip2"] = self.options.with_bzip2
         # requires LibXml2 cmake name
-        self._cmake.definitions["ENABLE_LIBXML2"] = self.options.with_libxml2
-        self._cmake.definitions["ENABLE_ICONV"] = self.options.with_iconv
-        self._cmake.definitions["ENABLE_EXPAT"] = self.options.with_expat
-        self._cmake.definitions["ENABLE_PCREPOSIX"] = self.options.with_pcreposix
+        tc.variables["ENABLE_LIBXML2"] = self.options.with_libxml2
+        tc.variables["ENABLE_ICONV"] = self.options.with_iconv
+        tc.variables["ENABLE_EXPAT"] = self.options.with_expat
+        tc.variables["ENABLE_PCREPOSIX"] = self.options.with_pcreposix
         if self.options.with_pcreposix:
-            self._cmake.definitions["POSIX_REGEX_LIB"] = "LIBPCREPOSIX"
-        self._cmake.definitions["ENABLE_LibGCC"] = False
-        self._cmake.definitions["ENABLE_CNG"] = self.options.with_cng
+            tc.variables["POSIX_REGEX_LIB"] = "LIBPCREPOSIX"
+        tc.variables["ENABLE_LibGCC"] = False
+        tc.variables["ENABLE_CNG"] = self.options.with_cng
         # turn off features
-        self._cmake.definitions["ENABLE_ACL"] = self.options.with_acl
+        tc.variables["ENABLE_ACL"] = self.options.with_acl
         # turn off components
-        self._cmake.definitions["ENABLE_TAR"] = False
-        self._cmake.definitions["ENABLE_CPIO"] = False
-        self._cmake.definitions["ENABLE_CAT"] = False
-        self._cmake.definitions["ENABLE_TEST"] = False
+        tc.variables["ENABLE_TAR"] = False
+        tc.variables["ENABLE_CPIO"] = False
+        tc.variables["ENABLE_CAT"] = False
+        tc.variables["ENABLE_TEST"] = False
         # too strict check
-        self._cmake.definitions["ENABLE_WERROR"] = False
-        if tools.Version(self.version) >= "3.4.2":
-            self._cmake.definitions["ENABLE_MBEDTLS"] = self.options.with_mbedtls
-        self._cmake.definitions["ENABLE_XATTR"] = self.options.with_xattr
-
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
-    def _patch_sources(self):
```

Comment:
not sure it was a good idea to move these replacement to patches, it's more work for every new version with the risk to have a bad patch.